src/__tests__/pages/Research.test.js

```diff
@@ -0,0 +1,135 @@
+import React from "react";
+import "@testing-library/jest-dom";
+import { screen, render } from "@testing-library/react";
+import Research, { BlogPostResults } from "../../pages/Research";
+import { BrowserRouter } from "react-router-dom";
+
+// Mock data for blog posts
+const blogPosts = [
+  {
+    title: "Beta-launch: PolicyEngine's enhanced microdata for policy analysis",
+    description:
+      "By integrating and calibrating multiple datasets, PolicyEngine makes world-class tax-benefit microsimulation modeling available to anyone.",
+    date: "2024-03-25 09:00:00",
+    tags: ["us", "technical"],
+    authors: ["max-ghenis", "nikhil-woodruff"],
+    filename: "enhanced-cps-beta.md",
+    image: "enhanced-cps-beta.png",
+  },
+  {
+    title:
+      "Child Tax Credit Provisions of the Tax Relief for American Workers and Families Act",
+    description:
+      "How the Wyden-Smith bipartisan tax package would affect families through 2025.",
+    date: "2024-03-25 09:00:00",
+    tags: ["us", "policy"],
+    authors: ["max-ghenis"],
+    filename: "trafwa-ctc.md",
+    image: "trafwa-ctc.webp",
+  },
+  {
+    title: "Individual income tax provisions of President Biden's 2025 Budget",
+    description:
+      "PolicyEngine projects the impact of expanding tax credits and raising taxes on high-income filers.",
+    date: "2024-03-27 09:00:00",
+    tags: ["us", "policy", "featured"],
+    authors: ["max-ghenis", "pavel-makarchuk"],
+    filename: "biden-budget-2025.md",
+    image: "biden-budget-2025.jpeg",
+  },
+  {
+    title: "PolicyEngine launches state income tax modeling nationwide",
+    description:
+      "Free, open-source platform now covers all 50 states and D.C., just in time for tax season.",
+    date: "2024-04-15 09:00:00",
+    tags: ["us", "policy", "featured"],
+    authors: ["pavel-makarchuk"],
+    filename: "state-tax-model-beta.md",
+    image: "state-tax-model-beta.png",
+  },
+  {
+    title: "PolicyEngine's Spring 2024 update",
+    description: "How PolicyEngine compares to the latest external forecasts.",
+    date: "2024-04-16 09:00:00",
+    tags: ["uk", "policy", "featured"],
+    image: "uk_march_update.png",
+    authors: ["nikhil-woodruff"],
+    filename: "obr-forecast-update-spring-24.ipynb",
+  },
+  {
+    title: "The Triple Lock Plus",
+    description:
+      "PolicyEngine estimates the impact of the UK government's proposed pension uprating policy reform.",
+    date: "2024-05-28 09:00:00",
+    tags: ["uk", "policy", "featured"],
+    authors: ["nikhil-woodruff"],
+    filename: "uk-triple-lock-plus.ipynb",
+    image: "uk-triple-lock-plus.jpg",
+  },
+];
```

Comment:
Could you add one UK post that isn't featured to this list, as well as one that's marked "global" with no country tag? Both of these are cases we might face. You could also probably drop `"PolicyEngine launches state income tax modeling nationwide"`, as it doesn't introduce any new permutation.

Reply:
Also, if you could add a state-level post (e.g., I think there's a New York one in there) and our sole Nigeria post, I think that would be beneficial.